frontends/p4/fromv1.0/programStructure.cpp

```diff
@@ -59,7 +59,14 @@ const IR::Annotations*
 ProgramStructure::addNameAnnotation(cstring name, const IR::Annotations* annos) {
     if (annos == nullptr)
         annos = IR::Annotations::empty;
-    return annos->addAnnotationIfNew(IR::Annotation::nameAnnotation, new IR::StringLiteral(name));
+    return annos->addAnnotationIfNew(IR::Annotation::nameAnnotation,
+                                     new IR::StringLiteral(name));
+}
+
+const IR::Annotations*
+ProgramStructure::addGlobalNameAnnotation(cstring name,
```

Comment:
The spec mandates a `globalname` annotation, which we have not used so far. It should imply a leading dot.
I don't know if we also allow `name` annotations with a leading dot, and if there is any difference.